cluster/terraform_kubernetes/filebeat.tf

```diff
@@ -0,0 +1,190 @@
+data "azurerm_key_vault_secret" "beats_url" {
+  name         = "BEATS-URL"
+  key_vault_id = data.azurerm_key_vault.key_vault.id
+}
+
+resource "kubernetes_service_account" "filebeat" {
+  metadata {
+    name      = "filebeat"
+    namespace = kubernetes_namespace.default_list["monitoring"].metadata[0].name
+    labels = {
+      "name"      = "filebeat"
+    }
+  }
+}
+
+
+resource "kubernetes_cluster_role" "filebeat" {
+  metadata {
+    name = "filebeat"
+    labels = {
+      "name"      = "filebeat"
+    }
+  }
+  rule {
+    api_groups = [""]
+    resources  = ["pods", "nodes", "namespaces"]
+    verbs      = ["get", "list", "watch"]
+  }
+
+    rule {
+    api_groups = ["apps"]
+    resources  = ["replicasets"]
+    verbs      = ["get", "list", "watch"]
+  }
+
+}
+resource "kubernetes_cluster_role_binding" "filebeat" {
+  metadata {
+    name = "filebeat"
+    labels = {
+      "name" = "filebeat"
+    }
+  }
+
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = kubernetes_cluster_role.filebeat.metadata[0].name
+  }
+
+  subject {
+    kind      = "ServiceAccount"
+    name      = "filebeat"
+    namespace = kubernetes_namespace.default_list["monitoring"].metadata[0].name
+  }
+}
+
+resource "kubernetes_config_map" "filebeat" {
+
+  metadata {
+    name      = "filebeat-config"
+    namespace = kubernetes_namespace.default_list["monitoring"].metadata[0].name
+  }
+
+  data = {
+    "filebeat.yml" = templatefile("${path.module}/config/prometheus/development.filebeat.yml.tmpl", local.filebeats_template_variable_map)
+  }
+
+}
+
+resource "kubernetes_daemonset" "filebeat" {
```

Comment:
Can this be deployed to the apps1 node pool only?

Reply:
added node selector